### PR TITLE
Cypress discussion

### PR DESCRIPTION
**Pull request by Kelketek**

### CODE REVIEW

File: common/test/acceptance/pages/lms/discussion.py [hunk 1]
```diff
@@ -419,6 +419,8 @@ def is_discussion_expanded(self):
 
     def expand_discussion(self):
         """Click the link to expand the discussion"""
+        import time
+        time.sleep(200)
```
Comment (e-kolpakov) [inline common/test/acceptance/pages/lms/discussion.py]: @Kelketek debug leftover

**Comment by e-kolpakov** [comment]
@Kelketek :+1:

File: common/lib/xmodule/xmodule/x_module.py [hunk 2]
```diff
@@ -1613,7 +1613,7 @@ def get_course_template_context(self):
         request.user = user
         user_info = cc.User.from_django_user(self.runtime.user).to_dict()
         course_id = self.runtime.course_id
-        course = get_course_with_access(self.runtime.user, 'load_forum', course_id)
+        course = get_course_with_access(self.runtime.user, 'load', course_id)
```
Comment (bradenmacdonald) [inline common/lib/xmodule/xmodule/x_module.py]: @Kelketek Do you think these also need to specify `check_if_enrolled`? See [here](https://github.com/edx/edx-platform/commit/d240785b17bdcaeb11e4ef24ee4a3697da26c9b4#diff-1f8b9b8344cecfd83d74693377466fd4) for why I ask.

File: lms/djangoapps/django_comment_client/forum/views.py [hunk 3]
```diff
@@ -148,7 +147,7 @@ def get_threads(request, course, discussion_id=None, per_page=THREADS_PER_PAGE):
             group_id = None
     else:
         group_id = get_cohort_id(request.user, course.id)
-        if not group_id and get_cohorted_threads_privacy(course.id) == 'cohort-only':
```
Comment (bradenmacdonald) [inline lms/djangoapps/django_comment_client/forum/views.py]: Was the cohort-only setting made mandatory, or did something else change here?

Comment (Kelketek) [inline lms/djangoapps/django_comment_client/forum/views.py]: Yes. The possibility of being outside of a cohort in a cohorted course was considered a bug and fixed by adding the 'default cohort'.

Comment (bradenmacdonald) [inline lms/djangoapps/django_comment_client/forum/views.py]: Ah, the change was that now all students should always be in a cohort.

File: common/lib/xmodule/xmodule/x_module.py [hunk 4]
```diff
@@ -1613,7 +1613,7 @@ def get_course_template_context(self):
         request.user = user
         user_info = cc.User.from_django_user(self.runtime.user).to_dict()
         course_id = self.runtime.course_id
-        course = get_course_with_access(self.runtime.user, 'load_forum', course_id)
+        course = get_course_with_access(self.runtime.user, 'load', course_id, check_if_enrolled=True)
         user_cohort_id = get_cohort_id(user, course_id)
 
         unsafethreads, query_params = get_threads(request, course_id)
```
Comment (bradenmacdonald) [inline common/lib/xmodule/xmodule/x_module.py]: This is passing in a `course_id` but the `get_threads` code seems to expect a `course` object (though it only really needs an ID).

As a result, I'm seeing:
```
  File "/edx/app/edxapp/edx-platform/common/lib/xblock/discussion/discussion_forum/discussion_forum.py", line 150, in student_view
    fragment = self._student_view_lms()
  File "/edx/app/edxapp/edx-platform/common/lib/xblock/discussion/discussion_forum/discussion_forum.py", line 160, in _student_view_lms
    context = discussion_service.get_course_template_context()
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1619, in get_course_template_context
    unsafethreads, query_params = get_threads(request, course_id)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic-2.46.0.37/newrelic/api/function_trace.py", line 104, in literal_wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/django_comment_client/forum/views.py", line 113, in get_threads
    'course_id': unicode(course.id),
AttributeError: 'CourseLocator' object has no attribute 'id'
```

Comment (bradenmacdonald) [inline common/lib/xmodule/xmodule/x_module.py]: Also, that code path should be covered by some kind of test.

File: common/lib/xmodule/xmodule/x_module.py [hunk 5]
```diff
@@ -1613,10 +1613,10 @@ def get_course_template_context(self):
         request.user = user
         user_info = cc.User.from_django_user(self.runtime.user).to_dict()
         course_id = self.runtime.course_id
-        course = get_course_with_access(self.runtime.user, 'load_forum', course_id)
+        course = get_course_with_access(self.runtime.user, 'load', course_id, check_if_enrolled=True)
         user_cohort_id = get_cohort_id(user, course_id)
 
-        unsafethreads, query_params = get_threads(request, course_id)
+        unsafethreads, query_params = get_threads(request, course)
```
Comment (bradenmacdonald) [inline common/lib/xmodule/xmodule/x_module.py]: This fake request object has no method, so it's causing a bug for me:

```
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1619, in get_course_template_context
    unsafethreads, query_params = get_threads(request, course)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/newrelic-2.46.0.37/newrelic/api/function_trace.py", line 104, in literal_wrapper
    return wrapped(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/django_comment_client/forum/views.py", line 116, in get_threads
    'group_id': get_group_id_for_comments_service(request, course.id, discussion_id),  # may raise ValueError
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/django_comment_client/utils.py", line 561, in get_group_id_for_comments_service
    if not requested_group_id:
UnboundLocalError: local variable 'requested_group_id' referenced before assignment
```

See `get_group_id_for_comments_service` in `utils.py` and you'll understand why. That code should probably treat either GET or POST as the default.

Comment (bradenmacdonald) [inline common/lib/xmodule/xmodule/x_module.py]: Also, having to use a fake request object outside of test code is usually a bad sign :-/

**Comment by bradenmacdonald** [comment]
Thanks for the quick fixes to those issues, @Kelketek .

:+1:

**Pull request merged**